pandas/core/arrays/base.py

```diff
@@ -610,3 +618,121 @@ def _ndarray_values(self):
         used for interacting with our indexers.
         """
         return np.array(self)
+
+
+class ExtensionOpsBase(object):
+    """
+    A base class for the mixins for different operators.
+    Can also be used to define an individual method for a specific
+    operator using the class method create_method()
+    """
+    @classmethod
+    def create_method(cls, op):
+        """
+        A class method that returns a method that will correspond to an
+        operator for an ExtensionArray subclass.
+
+        Parameters
+        ----------
+        op:   An operator that takes arguments op(a, b)
+
+        Returns
+        -------
+        A method that can be bound to a method of a class
+
+        Usage
+        -----
+        Given an ExtensionArray subclass called MyClass, use
+
+        mymethod = create_method(my_operator)
+        in the class definition of MyClass to create the operator
+
+        """
+        op_name = ops._get_op_name(op, False)
+
+        def _binop(self, other):
+            def convert_values(parm):
+                if isinstance(parm, ExtensionArray):
+                    ovalues = list(parm)
```

Comment:
Why is it needed to convert to a list? ExtensionArrays support iterating through them (which returns the scalars)

Reply:
Fixed